tests/altair_theme_test.py

```diff
@@ -0,0 +1,273 @@
+  <script type="text/javascript" src="{{ base_url }}/vega-lite@{{ vegalite_version }}"></script>
+  <script type="text/javascript" src="{{ base_url }}/vega-embed@{{ vegaembed_version }}"></script>
+</head>
+<body>
+  <div id="header">
+    Theme:
+    <select id="themes">
+      <option value="default">default</option>
+      <option value="excel">excel</option>
+      <option value="ggplot2">ggplot2</option>
+      <option value="quartz">quartz</option>
+      <option value="vox">vox</option>
+      <option value="dark">dark</option>
+      <option value="fivethirtyeight">fivethirtyeight</option>
+      <option value="latimes">latimes</option>
+      <option value="urbaninstitute">urbaninstitute</option>
+      <option value="googlecharts">googlecharts</option>
+      <option value="powerbi">powerbi</option>
+      <optgroup label="Carbon">
+        <option value="carbonwhite">carbonwhite</option>
+        <option value="carbong10">carbong10</option>
+        <option value="carbong90">carbong90</option>
+        <option value="carbong100">carbong100</option>
+      </optgroup>
+    </select>
+    <br />
+  </div>
+  <div id="{{ output_div }}"></div>
+  <script>
+    // Vega Theme Test - But single view, no renderer option
+    var spec = {{ spec }};
+    var container = document.querySelector("#{{ output_div }}");
+    var themes = document.querySelector("#themes");
+    var currentLocation = window.location;
+    var url = new URL(currentLocation);
+
+    themes.addEventListener("change", function () {
+      theme = themes.options[themes.selectedIndex].value;
+      url.searchParams.set("theme", `${themes.options[themes.selectedIndex].value}`);
+      window.history.replaceState(null, null, url);
+      refresh();
+    });
+
+    function refresh() {
+      const themeName = themes.options[themes.selectedIndex].value;
+      container.innerHTML = "";
+      var el = document.createElement("div");
+      el.setAttribute("class", "view");
+      container.appendChild(el);
+      vegaEmbed(el, spec, {
+          theme: themeName,
+          defaultStyle: true,
+          mode: "vega-lite"
+      });
+    }
+
+    // Mostly the standard template
+    // - Need to adapt these two ideas into one
+    (function(vegaEmbed) {
+      var embedOpt = { mode: "vega-lite" };
+
+      function showError(el, error){
+          el.innerHTML = ('<div style="color:red;">'
+                          + '<p>JavaScript Error: ' + error.message + '</p>'
+                          + "<p>This usually means there's a typo in your chart specification. "
+                          + "See the javascript console for the full traceback.</p>"
+                          + '</div>');
+          throw error;
+      }
+      const el = document.getElementById('{{ output_div }}');
+      vegaEmbed("#{{ output_div }}", spec, embedOpt)
+        .catch(error => showError(el, error));
+    })(vegaEmbed);
+
+  </script>
```

Comment:
Another solution could be parameterizing the template used in `altair-plot`. 

So an option to override `VGL_TEMPLATE`

https://github.com/vega/sphinxext-altair/blob/777d56e68e9134feab3b19202d5b03f752686d40/sphinxext_altair/altairplot.py#L96-L114

Essentially, working around the limitation highlighted in https://github.com/vega/altair/pull/3630#issuecomment-2408623223 by adding this refresh script in

Definitely not an elegant solution, but would give the greatest level of control.

Would look somewhat like the import reference in https://github.com/vega/altair/blob/42491c7dc5a6975441578b63c884ee74424cdbea/sphinxext/code_ref.py

But instead could either specify an inline `jinja2.Template` or maybe read one from a file?

Reply:
Abadoning in favor of `pyscript`, see https://github.com/vega/altair/pull/3630#issuecomment-2411420348